src/site/_includes/js/test/eligibility-programs.test.js

```diff
@@ -504,6 +504,23 @@ describe('Program eligibility', () => {
     });
   }
 
+  function testDischargeStatus(setupFn, resultFn, testCases) {
```

Comment:
you _could_ expand this test suite to also test eligibility.  At the moment, "Requires discharge that is not dishonorable" further down around line 1755 tests exactly that, but it is blind to whether or not eligibility is true for all the other discharge types.

If you test eligibility along with the complex discharge flag (like `testImmigration`) then you'll get full test coverage of all discharge options.

I consider this suggestion to be optional.